### PR TITLE
more file formats for IStreamQuadIterator

### LINKED ISSUE #164: Allow parsing Trig and NQuads
Currently, the parser is only able to parse Turtle (and its subset NTriple). The backend library, serd, is also able to Trig and NQuads. There should be an option to support those formats.

### PR DESCRIPTION
**Pull request by mcb5637**
resolve #164

### CODE REVIEW

File: src/rdf4cpp/rdf/parser/ParsingFlags.hpp [hunk 1]
```diff
@@ -9,6 +9,10 @@ namespace rdf4cpp::rdf::parser {
 enum struct ParsingFlag : uint8_t {
     Strict = 1 << 0,
     NoParsePrefix = 1 << 1,
+    Turtle = 1 << 2,
```
Comment (liss-h) [inline src/rdf4cpp/rdf/parser/ParsingFlags.hpp]: The flags could be encoded into the 2 bits after the `NoParsePrefix` bit. Behaviour would be essentially identical to how it is right now except that it saves some space (maybe relevant, idk yet) and it makes more sense semantically. Just encode turtle as `0b00` which automatically makes it the default value.

File: private/rdf4cpp/rdf/parser/IStreamQuadIteratorSerdImpl.hpp [hunk 2]
```diff
@@ -57,6 +57,18 @@ struct IStreamQuadIterator::Impl {
     static SerdStatus on_prefix(void *voided_self, SerdNode const *name, SerdNode const *uri) noexcept;
     static SerdStatus on_stmt(void *voided_self, SerdStatementFlags, SerdNode const *graph, SerdNode const *subj, SerdNode const *pred, SerdNode const *obj, SerdNode const *obj_datatype, SerdNode const *obj_lang) noexcept;
 
+    static constexpr SerdSyntax extract_syntax_from_flags(ParsingFlags flags) noexcept {
+        if (flags.contains(ParsingFlag::Turtle))
```
Comment (liss-h) [inline private/rdf4cpp/rdf/parser/IStreamQuadIteratorSerdImpl.hpp]: If you implement the thing from below, just take care to check the things in the correct order. Or alternatively make this into a switch on the flag's underlying repr with the non-syntax bits masked out (not that important tbh, you choose which of the options you like better).

**Review by liss-h — CHANGES_REQUESTED** [review]
Looks straightforward, just two comments related to the flags.
Btw, did you find out what the bug related to the quads actually was or was it just fixed by merging from serd's main repo?

**Comment by mcb5637** [comment]
was a bug in serd, link to the issue: https://github.com/drobilla/serd/issues/42

**Review by liss-h — APPROVED** [review]

**Pull request merged**